lib/modules/aws_ip_ranges.rb

```diff
@@ -0,0 +1,40 @@
+module AWSIpRanges
+  # Used based on this AWS instruction: https://forums.aws.amazon.com/ann.jspa?annID=2051
+  PATH = 'https://ip-ranges.amazonaws.com/ip-ranges.json'.freeze
+  COMPATIBLE_REGIONS = %w[GLOBAL eu-west-2].freeze
+
+  def self.cloudfront_ips
+    uri = URI(PATH)
+
+    begin
+      http_connection = Net::HTTP.new(uri.host, uri.port)
+      http_connection.read_timeout = 10
+      http_connection.open_timeout = 5
+      http_connection.use_ssl = true
+      response = http_connection.start { |http| http.get(uri.path) }
+
+      parse_json_for_ips(response.body)
+    rescue Timeout::Error,
+           Errno::EINVAL,
+           Errno::ECONNRESET,
+           EOFError,
+           Net::HTTPBadResponse,
+           Net::HTTPHeaderSyntaxError,
+           Net::ProtocolError => e
+      Rails.logger.warn("Unable to setup Rack Proxies to acquire the correct remote_ip: #{e.class}")
+      []
+    end
+  end
+
+  def self.parse_json_for_ips(response)
+    aws_ip_ranges = JSON.parse(response)
+
+    cloudfront_ip_blocks = aws_ip_ranges['prefixes'].select do |record|
+      COMPATIBLE_REGIONS.include?(record['region']) && record['service'] == 'CLOUDFRONT'
```

Comment:
I have also seen IP address in our logs which is not under service `CLOUDFRONT`
I have seen `64.252.83.x` which has service type `AMAZON` in the [json](https://ip-ranges.amazonaws.com/ip-ranges.json) document.
maybe the IP was rotated.

Reply:
I just tried pinging QA with `64.252.83.100` and it got filtered out - it looks like the AWS json list is suitable at the moment